### DISCUSSION TITLE
Consumption Calculation

### DISCUSSION THREAD

**Discussion by TheLordFolken** [wlcrs/huawei_solar#122] [Setting up the integration]
Has anybody found the path of least resistance to calculate the consumption of the load/house? I'm trying to graph my houses consumption in watts.

I tried to use the logic:
`
{{states('sensor.charge_discharge_power')|float+states('sensor.grid_active_power')|float+states('sensor.input_power')|float-states('sensor.grid_exported')|float}}
`
however charge_discharge_power is a negative value for discharging and a positive value for charging which I can't get my hear around in the above.

T

**Comment by JacoboSanchez** [comment]
You can use if conditions in the sensor definition but I prefer to split the calculation in simplier sensors to make it more readable later. For example you can create a discharge (or battery_to_house) sensor like this:
```
      battery_to_house:
        friendly_name: "Battery to house"
        unit_of_measurement: "W"
        device_class: power
        value_template: >-
          {% set u = states('sensor.charge_discharge_power') | int %}
          {% if u < 0 %}
              {{ (- u) | int }}
          {% else %}
              {{ (0) | int }}
          {% endif %}
        availability_template: >-
          {{ (states('sensor.charge_discharge_power') not in [ 'unavailable', 'none', 'unknown' ] ) }}
```

You have to keep in mind that you are making assumptions there. That definition is simple but does not cover the use case where you force the battery to discharge to grid (which I don't use so I am ok with that definition). 

There is a discussion with some samples on how to define power flow from one device to other [in this repo](https://github.com/reptilex/tesla-style-solar-power-card/issues/95) which may give you some ideas
